### PR TITLE
Coalesce `dep-info`, `dep-info-spaces` and `dep-info-doesnt-run-much` `run-make` tests into `dep-info` rmake.rs

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

This is one of the most ancient tests in the `run-make` directory and its Makefile does some unexpected things, like creating and deleting a `done` directory over and over, sleeping at certain times (this is the [commit](https://github.com/rust-lang/rust/pull/13288/commits/0d9fd8e2a1f8aa43b4cf66272eaa3cf695bbdcd9) that added the `sleep`).

I tried to preserve the intent of the test, which is smoke-testing that `dep-info` works.

try-job: x86_64-msvc
try-job: i686-mingw
try-job: aarch64-gnu
try-job: aarch64-apple
try-job: test-various
try-job: armhf-gnu
try-job: dist-various-1

### CODE REVIEW

File: tests/run-make/dep-info/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// Then, it checks that compilation using the flag is successful in general, even with
+// empty source files or source files that contain a whitespace character.
+
+// Finally, it removes one dependency and checks that compilation is still successful.
+// See https://github.com/rust-lang/rust/pull/10698
+
+// FIXME try it on windows and freebsd
+// on windows `rustc --dep-info` produces Makefile dependency with
+// windows native paths (e.g. `c:\path\to\libfoo.a`)
+// but msys make seems to fail to recognize such paths, so test fails.
```
Comment (jieyouxu) [inline tests/run-make/dep-info/rmake.rs:15]: Remark: in msys2, the path might actually be a unix path... so our host (`uname -a`) is

```bash
$ uname -a
MSYS_NT-10.0-22631 Joe 3.5.3.x86_64 2024-06-03 06:22 UTC x86_64 Msys
```
 
Screenshot 2024-08-11 111003

**Review by jieyouxu — APPROVED** [review]

**Review by jieyouxu — APPROVED** [review]
Problem: we might actually need a conditional dependency on some kind of `make` here to preserve what the actual test intends to check: that `rustc --dep-info` outputs some Makefile-compatible file that can be successfully interpreted by some `make`. I'm thinking

```
//@ needs-make
```

or something insane like that...

**Review by jieyouxu — CHANGES_REQUESTED** [review]
Sorry I clicked wrong button

File: tests/run-make/dep-info-spaces/Makefile.foo [hunk 2]
```diff
@@ -1,7 +0,0 @@
-LIB := $(shell $(RUSTC) --print file-names --crate-type=lib $(TMPDIR)/lib.rs)
-
-$(TMPDIR)/$(LIB):
-	$(RUSTC) --emit link,dep-info --crate-type=lib $(TMPDIR)/lib.rs
-	touch $(TMPDIR)/done
-
--include $(TMPDIR)/lib.d
```
Comment (jieyouxu) [inline tests/run-make/dep-info-spaces/Makefile.foo:7]: Actually I stand corrected, this doesn't even check `lib.d` is valid Makefile syntax, so I'm fine with what the test does as-is.

**Comment by Oneirical** [comment]
I'm not sure if you want me to change something in particular. I removed the ignore statement, since the try job passed. @rustbot review

File: tests/run-make/dep-info/rmake.rs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// This is a simple smoke test for rustc's `--emit dep-info` feature. It prints out
+// information about dependencies in a Makefile-compatible format, as a `.d` file.
```
Comment (jieyouxu) [inline tests/run-make/dep-info/rmake.rs:2]: Suggestion: we should make it clear that we don't attempt to check the `.d` file is in fact Makefile-compatible in this test.

**Comment by jieyouxu** [comment]
Feel free to r=me after clarifying the comment and if try jobs come back green.
@bors delegate+

File: tests/run-make/dep-info/rmake.rs (hunk 3 above)
Comment (jieyouxu) [reply tests/run-make/dep-info/rmake.rs:2]: @rustbot author

**Comment by jieyouxu** [comment]
bors go home you're drunk

**Pull request merged**